crates/re_log_types/src/path/parse_path.rs

```diff
@@ -22,25 +29,243 @@ pub enum PathParseError {
 
     #[error("Missing slash (/)")]
     MissingSlash,
+
+    #[error("Extra trailing slash (/)")]
+    TrailingSlash,
+
+    #[error("Invalid character: {character:?} in entity path identifier {part:?}. Only ASCII characters, numbers, underscore, and dash are allowed. To put wild text in an entity path, surround it with double-quotes.")]
+    InvalidCharacterInPart { part: String, character: char },
+
+    #[error("Invalid instance key: {0:?} (expected '[#1234]')")]
+    BadInstanceKey(String),
+
+    #[error("Found an unexpected instance key: [#{}]", 0.0)]
+    UnexpectedInstanceKey(InstanceKey),
+
+    #[error("Found an unexpected trailing component name: {0:?}")]
+    UnexpectedComponentName(ComponentName),
+
+    #[error("Found no component name")]
+    MissingComponentName,
+
+    #[error("Found trailing dot (.)")]
+    TrailingDot,
 }
 
-/// Parses an entity path, e.g. `foo/bar/#1234/5678/"string index"/a6a5e96c-fd52-4d21-a394-ffbb6e5def1d`
-pub fn parse_entity_path(path: &str) -> Result<Vec<EntityPathPart>, PathParseError> {
-    if path.is_empty() {
-        return Err(PathParseError::EmptyString);
+type Result<T, E = PathParseError> = std::result::Result<T, E>;
+
+impl std::str::FromStr for DataPath {
+    type Err = PathParseError;
+
+    /// For instance:
+    ///
+    /// * `world/points`
+    /// * `world/points.Color`
+    /// * `world/points[#42]`
+    /// * `world/points[#42].rerun.components.Color`
+    fn from_str(path: &str) -> Result<Self, Self::Err> {
```

Comment:
am I missing something or is there no test for `DataPath` parsing itself, only for the smaller subparts?
Can we at least add something that tests the "complex" case of `world/points[#42].rerun.components.Color`?